wrangler-service/src/main/java/io/cdap/wrangler/service/schema/SchemaViewerHandler.java

```diff
@@ -0,0 +1,219 @@
+   * @param id The id of the standard returned by the .../schemas listing.
+   */
+  @GET
+  @Path("contexts/{context}/schemaviewer/schemas/{id}")
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  public void getSchema(HttpServiceRequest request, HttpServiceResponder responder,
+      @PathParam("context") String namespace, @PathParam("id") String id) {
+    respond(request, responder, namespace, ns -> TransactionRunners.run(getContext(), context -> {
+      SchemaFileStore store = getStore(context, id);
+      return new ServiceResponse<>(store.readSchemaFileDescriptor(id));
+    }));
+  }
+
+  /**
+   * Returns all of the top level entities associated with a specific standard.
+   *
+   * @param request Handler for incoming request.
+   * @param responder Responder for data going out.
+   * @param namespace Namespace of request.
+   * @param id The id of the standard returned by the .../schemas listing.
+   */
+  @GET
+  @Path("contexts/{context}/schemaviewer/schemas/{id}/entities")
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  public void getSchemaEntities(HttpServiceRequest request, HttpServiceResponder responder,
+      @PathParam("context") String namespace, @PathParam("id") String id) {
+    respond(request, responder, namespace, ns -> TransactionRunners.run(getContext(), context -> {
+      SchemaFileStore store = getStore(context, id);
+      List<String> entities = new ArrayList<>();
+      Discriminator discriminator = store.readDiscriminator(id);
+      if (discriminator != null && discriminator.getMapping() != null) {
+        entities.addAll(discriminator.getMapping().keySet());
+      }
+      return new ServiceResponse<>(entities);
+    }));
+  }
+
+  /**
+   * Returns the JSON schema definition for a top level entity within a standard.
+   *
+   * @param request Handler for incoming request.
+   * @param responder Responder for data going out.
+   * @param namespace Namespace of request.
+   * @param id The id of the standard returned by the .../schemas listing.
+   * @param name The name of the entity.
+   */
+  @GET
+  @Path("contexts/{context}/schemaviewer/schemas/{id}/entities/{name}")
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  public void getEntitySchemaDefinition(HttpServiceRequest request, HttpServiceResponder responder,
+      @PathParam("context") String namespace,
+      @PathParam("id") String id,
+      @PathParam("name") String name) {
+    respond(request, responder, namespace, ns -> TransactionRunners.run(getContext(), context -> {
+      SchemaFileStore store = getStore(context, id);
+      SchemaFileDescriptor descriptor = store.readSchemaFileDescriptor(id);
+      if (descriptor.getSchemaFile() == null) {
+        throw new NotFoundException(String.format(
+            "There are no entity schema definitions for standard %s .", id));
+      }
+      Map<String, JsonElement> definitions = descriptor.getSchemaFile().getDefinitions();
```

Comment:
similar comment here, where the logic should be pushed down instead of in the handler